src/Language/Haskell/Liquid/Types.hs

```diff
@@ -458,8 +458,9 @@ instance Hashable (PVar a) where
 ------ Strictness --------------------------------------------------
 --------------------------------------------------------------------
 
-instance NFData Var
-instance NFData SrcSpan
+instance NFData Var where rnf _ = ()
```

Comment:
If I remember correctly, -fwarn-unused-matches and -fwarn-unused-binds have
caught a few bugs in the past. We should at least keep those around.

On Fri, Mar 27, 2015 at 4:45 PM, Eric Seidel <notifications@github.com>
wrote:

> In src/Language/Haskell/Liquid/Types.hs
> <https://github.com/ucsd-progsys/liquidhaskell/pull/364#discussion_r27338863>
> :
>
> > @@ -458,8 +458,9 @@ instance Hashable (PVar a) where
> >  ------ Strictness --------------------------------------------------
> >  --------------------------------------------------------------------
> >
> > -instance NFData Var
> > -instance NFData SrcSpan
> > +instance NFData Var where rnf _ = ()
>
> I take that back. Looking at the list of warnings that GHC supports, the
> standard warnings seem completely reasonable. However, the additional
> warnings enabled by -W (-fwarn-unused-binds, -fwarn-unused-matches,
> -fwarn-unused-imports, -fwarn-incomplete-patterns, -fwarn-dodgy-exports,
> -fwarn-dodgy-imports) don't seem particularly useful with the exception of
> incomplete-patterns.
>
> Incidentally, most of the new warnings are only triggered by -W, so maybe
> we can just remove that :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ucsd-progsys/liquidhaskell/pull/364/files#r27338863>.
>